components/org.wso2.carbon.secvault/src/main/java/org/wso2/carbon/secvault/internal/SecureVaultActivator.java

```diff
@@ -39,14 +39,21 @@
 
     @Override
     public void start(BundleContext bundleContext) throws Exception {
-        SecureVaultDataHolder.getInstance().setBundleContext(bundleContext);
-        logger.debug("Starting Secure Vault bundle");
-        logger.debug("Initializing Secure Vault config...");
-        Path secureVaultYAMLPath = Utils.getRuntimeConfigPath().resolve(Constants.DEPLOYMENT_CONFIG_YAML);
-        SecureVaultDataHolder.getInstance().setSecureVaultConfiguration(SecureVaultUtils.getSecureVaultConfig
-                (secureVaultYAMLPath).orElseThrow(() -> new SecureVaultException("Error occurred when obtaining " +
-                "secure vault configuration.")));
-        logger.debug("Secure vault config successfully initialized");
+        try {
+            SecureVaultDataHolder.getInstance().setBundleContext(bundleContext);
+            logger.debug("Starting Secure Vault bundle");
+            logger.debug("Initializing Secure Vault config...");
+            Path secureVaultYAMLPath = Utils.getRuntimeConfigPath().resolve(Constants.DEPLOYMENT_CONFIG_YAML);
+            SecureVaultDataHolder.getInstance().setSecureVaultConfiguration(SecureVaultUtils.getSecureVaultConfig
+                    (secureVaultYAMLPath).orElseThrow(() -> new SecureVaultException("Error occurred when obtaining " +
+                    "secure vault configuration.")));
+            logger.debug("Secure vault config successfully initialized");
+        } catch (Throwable throwable) {
```

Comment:
Also, do we really need to catch `Throwable` here. Any reason why `Exception` (or any other exception) is not sufficient.

Reply:
Why we shouldn't catch `Throwable`: https://stackoverflow.com/a/6083271/1577286

Reply:
Fixes apply: ff567a883e26a15f718baf7c23981ae4fffa4697

Reply:
Reg: Throwable, yes it may be best practise to catch the Exception only. But here the problem is there is issue in carbon logging framework where it coudn't catch the eclipse equinox frameowrk logs and becouse of the it may hidden the Throwable causes without logging to carbon log file.